### PR TITLE
Clean Up MyTBA Page

### PR DESCRIPTION
**Pull request by phil-lopreiato**
Now shows upcoming events as well!
screenshot from 2016-04-21 19-00-38

### CODE REVIEW

**Comment by phil-lopreiato** [comment]
Added button to view the live event webcasts, if they exist

screenshot from 2016-04-21 19-30-24

**Comment by gregmarra** [comment]
Nice! The alignment on "upcoming competitions" vs other headers indicates somethings funky w your div structure here.

For one day events, do we have a way of showing just the single day? See Einstein in your ss.

File: common/my_filters.py [hunk 1]
```diff
@@ -35,3 +35,8 @@ def yt_start(value):
 @register.filter
 def get_item(dictionary, key):
     return dictionary.get(key)
+
+
+@register.filter
+def strftime(d, str):
```
Comment (fangeugene) [inline common/my_filters.py]: You should be able to use this instead.

https://docs.djangoproject.com/en/1.9/ref/templates/builtins/#date

**Comment by phil-lopreiato** [comment]
Good catch, I'll fix that wonky alignment. There's probably a way to fix the date there, I just copied this code from the team detail page. But this definitely worth a fix in both locations

**Comment by fangeugene** [comment]
Re @gregmarra 's comment about one day events, look at how the event details template does it.

File: controllers/mytba_controller.py [hunk 2]
```diff
@@ -19,21 +21,38 @@ def get(self):
...
 
         live_events = EventHelper.getWeekEvents()
+        all_events_this_year_future = EventListQuery(now.year).fetch_async()
         favorite_team_keys = map(lambda f: ndb.Key(Team, f.model_key), team_favorites_future.get_result())
         favorite_teams_future = ndb.get_multi_async(favorite_team_keys)
 
+        future_events = filter(lambda e: e.start_date > now, all_events_this_year_future.get_result())
+        future_eventteams_futures = []
+        for event in future_events:
+            future_eventteams_futures.append(EventTeamsQuery(event.key_name).fetch_async())
```
Comment (fangeugene) [inline controllers/mytba_controller.py]: Is this fetching eventteams for all future events and then filtering for favorites? That sounds insanely inefficient. Perhaps try querying for eventteams of favorite teams, then querying for events that are in the future?

Comment (phil-lopreiato) [inline controllers/mytba_controller.py]: That approach won't be able to use `DatabaseQuery`s though. I don't have a good grasp of the performance differences - I'll do whichever approach you think is better.

Comment (fangeugene) [inline controllers/mytba_controller.py]: How about making a TeamEvents DatabaseQuery for each team. That's one db read per team, which you can't really expect to get any better. Then filter the events accordingly by time.

Comment (phil-lopreiato) [inline controllers/mytba_controller.py]: There already is a TeamEvents query, but it fetch all events over all years. Is that good enough to use?

Comment (fangeugene) [inline controllers/mytba_controller.py]: There's a TeamYearEvents query

I guess efficiently compared to your original methods still depends on if the average user have more favorites than the average number of future events.

Comment (phil-lopreiato) [inline controllers/mytba_controller.py]: True. Which would you prefer?

Comment (fangeugene) [inline controllers/mytba_controller.py]: I'm going to say do the TeamYearEvents query on all favorites, then filter.

I think the average user has less than 50 favorites, and filtering each team's events for the next one is less work than filtering all events' event teams for all favorite teams.

**Comment by phil-lopreiato** [comment]
Updated screenshot

screenshot from 2016-04-21 20-55-22

**Comment by nwalters512** [comment]
Alignment is still weird...

**Comment by phil-lopreiato** [comment]
Let's try again...
screenshot from 2016-04-22 11-43-37

File: controllers/mytba_controller.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import collections
```
Comment (fangeugene) [inline controllers/mytba_controller.py]: Unused import.

File: controllers/mytba_controller.py [hunk 4]
```diff
@@ -6,9 +7,11 @@
 
 from consts.model_type import ModelType
 from controllers.base_controller import LoggedInHandler
+from database.event_query import EventListQuery, TeamYearEventsQuery
```
Comment (fangeugene) [inline controllers/mytba_controller.py]: `EventListQuery` unused

File: controllers/mytba_controller.py [hunk 5]
```diff
@@ -19,6 +21,7 @@ def get(self):
         self._require_registration('/account/register')
 
         user = self.user_bundle.account.key
+        now = datetime.datetime.now()
         team_favorites_future = Favorite.query(Favorite.model_type == ModelType.TEAM, ancestor=user).fetch_async()
 
         live_events = EventHelper.getWeekEvents()
```
Comment (fangeugene) [inline controllers/mytba_controller.py]: Perhaps change this to `getEventsWithinADay()` so a team doesn't show up in this list as well as the upcoming list? You might need to change the logic below to:

`e for e in events if e.start_date > now and not e.within_a_day()`

Comment (phil-lopreiato) [inline controllers/mytba_controller.py]: Good catch, I'll update the logic

**Pull request merged**